pkg/apiserver/certificate/cacert_controller.go

```diff
@@ -130,39 +137,48 @@ func (c *CACertController) syncCACert() error {
 // syncMutatingWebhooks updates the CABundle of the MutatingWebhookConfiguration backed by antrea-controller.
 func (c *CACertController) syncMutatingWebhooks(caCert []byte) error {
 	klog.Info("Syncing CA certificate with MutatingWebhookConfigurations")
-	for _, name := range c.caConfig.MutationWebhooks {
-		mWebhook, err := c.client.AdmissionregistrationV1().MutatingWebhookConfigurations().Get(context.TODO(), name, metav1.GetOptions{})
-		if err != nil {
-			return fmt.Errorf("error getting MutatingWebhookConfiguration %s: %v", name, err)
-		}
-		err = c.patchWebhookWithCACert(mWebhook, caCert)
-		if err != nil {
-			return fmt.Errorf("error updating antrea CA cert of MutatingWebhookConfiguration %s: %v", name, err)
-		}
+	mWebhooks, err := c.client.AdmissionregistrationV1().MutatingWebhookConfigurations().List(context.TODO(), antreaListOption)
+	if err != nil {
+		return fmt.Errorf("error listing antrea MutatingWebhookConfiguration: %v", err)
 	}
-	for _, name := range c.caConfig.OptionalMutationWebhooks {
-		mWebhook, err := c.client.AdmissionregistrationV1().MutatingWebhookConfigurations().Get(context.TODO(), name, metav1.GetOptions{})
-		if err != nil {
-			if errors.IsNotFound(err) {
-				klog.V(2).Infof("Optional mutation webhook %s not found, skipping its update", name)
-				continue
+	supportedMutatingWebhooks := sets.NewString(c.caConfig.MutationWebhooks...)
+	supportedOptionalWebhooks := sets.NewString(c.caConfig.OptionalMutationWebhooks...)
+	unSupporteWebhooks := sets.NewString()
+	for _, mWebhook := range mWebhooks.Items {
+		name := mWebhook.Name
+		if supportedMutatingWebhooks.Has(name) {
+			err = c.patchWebhookWithCACert(&mWebhook, caCert)
+			if err != nil {
+				return fmt.Errorf("error updating antrea CA cert of MutatingWebhookConfiguration %s: %v", name, err)
 			}
-			return fmt.Errorf("error getting MutatingWebhookConfiguration %s: %v", name, err)
-		}
-		err = c.patchWebhookWithCACert(mWebhook, caCert)
-		if err != nil {
-			return fmt.Errorf("error updating antrea CA cert of MutatingWebhookConfiguration %s: %v", name, err)
+		} else if supportedOptionalWebhooks.Has(name) {
+			err = c.patchWebhookWithCACert(&mWebhook, caCert)
+			if err != nil {
+				return fmt.Errorf("error updating antrea CA cert of MutatingWebhookConfiguration %s: %v", name, err)
+			}
+		} else {
+			unSupporteWebhooks.Insert(name)
```

Comment:
If the `antreaListOption` is added to get mutate webhook, will there be any unSupporteWebhooks?

Reply:
I originally add this check with "unSupporteWebhooks" because I have a concern that the deployed Webhooks include the resources not supported by antrea-controller logic, e.g., user added a wrong webhook in the manifest although it is added with a label "app: antrea". Then we could get some clues from this log. However, it seems a unnecessary concern?